adapters/base/src/main/java/org/apache/cassandra/sidecar/adapters/base/data/SessionInfo.java

```diff
@@ -0,0 +1,160 @@
+{
+    public final String peer;
+    public final int sessionIndex;
+    public final String connecting;
+    /** Immutable collection of receiving summaries */
+    public final List<StreamSummary> receivingSummaries;
+    /** Immutable collection of sending summaries*/
+    public final List<StreamSummary> sendingSummaries;
+    /** Current session state */
+    public final String state;
+    public final List<ProgressInfo> receivingFiles;
+    public final List<ProgressInfo> sendingFiles;
+
+    public SessionInfo(CompositeData data)
+    {
+        this.peer = (String) data.get("peer");
+        this.sessionIndex = (int) data.get("sessionIndex");
+        this.connecting = (String) data.get("connecting");
+        this.receivingSummaries = parseSummaries((CompositeData[]) data.get("receivingSummaries"));
+        this.sendingSummaries = parseSummaries((CompositeData[]) data.get("sendingSummaries"));
+        this.state = (String) data.get("state");
+        this.receivingFiles = parseFiles((CompositeData[]) data.get("receivingFiles"));
+        this.sendingFiles = parseFiles((CompositeData[]) data.get("sendingFiles"));
+    }
+
+    /**
+     * @return total size(in bytes) already received.
+     */
+    public long totalSizeReceived()
+    {
+        return totalSizeInProgress(receivingFiles);
+    }
+
+    /**
+     * @return total size(in bytes) already sent.
+     */
+    public long totalSizeSent()
+    {
+        return totalSizeInProgress(sendingFiles);
+    }
+
+    /**
+     * @return total number of files to receive in the session
+     */
+    public long totalFilesToReceive()
+    {
+        return totalFiles(receivingSummaries);
+    }
+
+    /**
+     * @return total number of files to send in the session
+     */
+    public long totalFilesToSend()
+    {
+        return totalFiles(sendingSummaries);
+    }
+
+    /**
+     * @return total size(in bytes) to receive in the session
+     */
+    public long totalSizeToReceive()
+    {
+        return totalSizes(receivingSummaries);
+    }
+
+    /**
+     * @return total size(in bytes) to send in the session
+     */
+    public long totalSizeToSend()
+    {
+        return totalSizes(sendingSummaries);
+    }
+
+    /**
+     * @return total number of files already received.
+     */
+    public long totalFilesReceived()
+    {
+        return totalFilesCompleted(receivingFiles);
+    }
+
+    /**
+     * @return total number of files already sent.
+     */
+    public long totalFilesSent()
+    {
+        return totalFilesCompleted(sendingFiles);
+    }
+
+    private long totalSizes(List<StreamSummary> summaries)
+    {
+        long total = 0;
+        for (StreamSummary summary : summaries)
+            total += summary.totalSize;
+        return total;
+    }
+
+    private long totalFilesCompleted(List<ProgressInfo> files)
+    {
+        Iterable<ProgressInfo> completed = Iterables.filter(files, input -> input.isCompleted());
+        return Iterables.size(completed);
```

Comment:
maybe simpler to do ? Any reason why we decide to use Iterables here?
```suggestion
                return files.stream()
                    .filter(ProgressInfo::isCompleted)
                    .count();
```